src/backup-all-indexes.sh

```diff
@@ -87,9 +95,10 @@ curl -w "\n" -sS -XPUT ${REPOSITORY_URL} -d "{
     \"server_side_encryption\": true
   }
 }"
-
-CUTOFF_DATE=$(date --date="${MAX_DAYS_TO_KEEP} days ago" +"%Y.%m.%d")
-echo "$(now) Archiving all indexes with logs before ${CUTOFF_DATE}."
+ARCHIVE_CUTOFF_DATE=$(date --date="1 days ago" +"%Y.%m.%d") # why would we wait weeks to archive?
```

Comment:
More importantly: I'm a little concerned about using exactly "1 day" as the cutoff. Using exactly one day sort of assumes that everybody agrees on what the date is — i.e. that they use the same timezone and their time is in sync —, which may not always be 100% accurate.

Unfortunately, [as I noted in this other PR](https://github.com/aptible/elasticsearch-logstash-s3-backup/pull/16/files#r144498117), snapshotting too early in this script means  you may not end up snapshotting everything...

I figure we could either create a final snapshot before deleting an index, _or_ give it 2 days before snapshotting..?

Reply:
Yeah, 2 days is good, but I'm unclear how the container running this script and the container the ES data is in would disagree.

My main concern is having ES die, and losing 7, 14, 30, whatever days of data because the archive waits until delete.

Reply:
> Yeah, 2 days is good, but I'm unclear how the container running this script and the container the ES data is in would disagree.

If you're assuming this is running on Enclave, then yes, this is unlikely to happen since we're running Logstash using UTC as the timezone, and you're using this container which also uses UTC as the timezone.

However, that might not be the case for everyone using this: if someone runs Logstash in UTC, but executes this script from a different TZ, they might end up backing up their logs too early because the script thinks today is day N+1 (and thus backs up the logstash-N index), but Logstash thinks today is day N and continues to write to the logstash-N index, thus resulting in any logs written after the backup script runs to never be archived.

> My main concern is having ES die, and losing 7, 14, 30, whatever days of data because the archive waits until delete.

Worth mentioning that, if you're assuming this is running on Enclave, this is largely moot considering the ES disk is actually backed up on a daily basis, though I can understand that you wanting to have your own archives.